include/seqan3/search/search_result.hpp

```diff
@@ -81,22 +81,53 @@ class search_result
+     * \tparam query_id_t Must be equal to query_id_t, if query_id_t is not empty_type.
+     * \tparam cursor_t Must be equal to cursor_t, if cursor_t is not empty_type.
+     * \tparam reference_id_t Must be equal to reference_id_t, if reference_id_t is not empty_type.
+     * \tparam reference_begin_pos_t Must be equal to reference_begin_pos_t, if reference_begin_pos_t is not empty_type.
      * \param[in] q_id The query id of the search result.
+     * \param[in] cursor The cursor of the search result.
      * \param[in] ref_id The reference id of the search result.
      * \param[in] ref_pos The reference begin position of the search result.
+     *
+     * Each member (the order matters!) will only be initialized if the respective type is not empty_type.
+     * That way, the constructor call is independent of the output configuration and the search result
+     * will handle the initialisation.
      */
-    search_result(query_id_type q_id, reference_id_type ref_id, reference_begin_pos_type ref_pos) :
-        query_id_(q_id), reference_id_(std::move(ref_id)), reference_begin_pos_(std::move(ref_pos))
-    {}
+    template <typename query_id_t, typename cursor_t, typename reference_id_t, typename reference_begin_pos_t>
+    search_result(query_id_t q_id,
+                  [[maybe_unused]] cursor_t cursor,
+                  [[maybe_unused]] reference_id_t ref_id,
+                  [[maybe_unused]] reference_begin_pos_t ref_pos)
+    {
+        if constexpr (!std::same_as<query_id_type, detail::empty_type>)
+        {
+            static_assert(std::assignable_from<query_id_type &, query_id_t &&>,
+                          "The member variable query_id_ must be assignable from parameter q_id!");
+            query_id_ = std::move(q_id);
+        }
+        if constexpr (!std::same_as<cursor_type, detail::empty_type>)
+        {
+            static_assert(std::assignable_from<cursor_type &, cursor_t &&>,
+                          "The member variable cursor_ must be assignable from parameter cursor!");
+            cursor_ = std::move(cursor);
+        }
+        if constexpr (!std::same_as<reference_id_type, detail::empty_type>)
+        {
+            static_assert(std::assignable_from<reference_id_type &, reference_id_t &&>,
+                          "The member variable passed reference_id_ must be assignable from parameter ref_id!");
+            reference_id_ = std::move(ref_id);
+        }
+        if constexpr (!std::same_as<reference_begin_pos_type, detail::empty_type>)
+        {
+            static_assert(std::assignable_from<reference_begin_pos_type &, reference_begin_pos_t &&>,
+                          "The member variable passed to reference_begin_pos_ must be assignable from parameter "
+                          "ref_pos!");
+            reference_begin_pos_ = std::move(ref_pos);
+        }
+        (void) q_id; // for some reason the ctor doesn't acceppt [[maybe_unused]] for q_id?!
```

Comment:
Can you investigate?

Reply:
I have no clue. Putting all parameters on one line also breaks, changing the order to `type [[maybe_unused]] val` makes the compiler ignore the attribute. I don't know how to further investigate this?

Reply:
Edit: `type val [[maybe_unused]]` works https://godbolt.org/z/vAkqE4
works but then doxygen has problems finding the wuery_id parameter :sweat: :sweat: 
What to do?

Reply:
bug report at gcc?

Reply:
This is a known bug fixed for 9.3+ https://gcc.gnu.org/bugzilla/show_bug.cgi?id=81429